core/lib/spree/core/null_promotion_configuration.rb

```diff
@@ -13,6 +13,14 @@ class NullPromotionConfiguration < Spree::Preferences::Configuration
       #   the standard coupon code handler class
       #   Spree::NullPromotionHandler.
       class_name_attribute :coupon_code_handler_class, default: 'Spree::NullPromotionHandler'
+
+      # Allows providing a different promotion finder.
+      # @!attribute [rw] promotion_finder_class
+      # @see Spree::NullPromotionFinder
+      # @return [Class] an object that conforms to the API of
+      #   the standard promotion finder class
+      #   Spree::NullPromotionFinder.
+      class_name_attribute :promotion_finder_class, default: 'Spree::NullPromotionFinder'
```

Comment:
I probably missed or forgot something in the past PRs but I don't understand how and where are we using this class (NullPromotionConfiguration). Can you please refresh my mind?

Reply:
Sure. Once the legacy promotion system is out of core, we have no more `Spree::Promotion` in core. All the touch points between the core system will be in a class called `Spree::Core::PromotionConfiguration`. However, even that class contains references to the legacy promotion system, so core will ship with a null object, the `Spree::NullPromotionConfiguration`. That class serves as documentation as to what a promotion configuration needs to provide, and is a stand-in that provides a bunch of service objects that do nothing but help us test the interaction between core and the configured promotion system.

Reply:
Nice, so the idea is that each "adapter" ships with its own configuration object, and what Solidus does, is using the right configuration class to understand which promotion handler to use, right?  Shouldn't we have a `Spree::LegacyPromotionConfiguration` somewhere as well?

Reply:
`Spree::Core::PromotionConfiguration` is the `Spree::LegacyPromotionConfiguration`. I can rename it once it's moved to the `legacy_promotions` gem, but I would prefer to keep that name as the legacy_promotions gem doesn't really have `legacy` in its namespace in order to keep the constants the same once the legacy_promotions gem is loaded.

Reply:
👍 Makes sense, thanks Martin!